src/Symfony/Component/Notifier/Bridge/Infobip/InfobipTransportFactory.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Infobip;
+
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Jérémy Romey <jeremy@free-agent.fr>
+ *
+ * @experimental in 5.1
+ */
+final class InfobipTransportFactory extends AbstractTransportFactory
+{
+    /**
+     * @return InfobipTransport
+     */
+    public function create(Dsn $dsn): TransportInterface
+    {
+        $scheme = $dsn->getScheme();
+        $username = $this->getUser($dsn);
+        $password = $this->getPassword($dsn);
+        $from = $dsn->getOption('from');
```

Comment:
You should probably throw an exception if not passed (have a look at the Freemobile transport  factory for an example).

Reply:
I've added it but we do not have that exception thrown for Nexmo and Twilio. Do you want me to add it too for those?